pkg/target/regolib/src.rego

```diff
@@ -0,0 +1,161 @@
+package target
+
+##################
+# Required Hooks #
+##################
+
+matching_constraints[constraint] {
+	constraint := data["{{.ConstraintsRoot}}"][_][_]
+	groups := {input.review.kind.group, "*"}
+	spec := get_default(constraint, "spec", {})
+	match := get_default(spec, "match", {})
+	kindSelector := get_default(match, "kinds", [{"apiGroups": ["*"], "kinds": ["*"]}])
+	selected_groups := {g | g = kindSelector[r].apiGroups[_]}
+	remaining_groups := groups - selected_groups
+	count(remaining_groups) < 2
+
+	kinds := {input.review.kind.kind, "*"}
+	selected_kinds := {k | k = kindSelector[r].kinds[_]}
```

Comment:
@maxsmythe I think you intended `r` to bind to the same value from the comprehension above but not that's going to happen. Here's an alternative implementation of the kind selector matching that gives us the semantics that (I think) we want: https://play.openpolicyagent.org/p/DpckwavJZE. LMKWYT.

Reply:
As a follow-up. It would be good to have some tests in place for the kind selector matching.

Reply:
Thank you for the alternative kind selector implementation!

I folded a slightly modified version of your code into the latest patch.